lib/fluent/plugin/in_tail.rb

```diff
@@ -213,8 +219,30 @@ def expand_paths
           paths << path
         end
       }
-      excluded = @exclude_path.map { |path| path = date.strftime(path); path.include?('*') ? Dir.glob(path) : path }.flatten.uniq
-      paths - excluded
+      excluded = @exclude_path.map {|path| path = date.strftime(path); path.include?('*') ? Dir.glob(path) : path}.flatten.uniq
+      to_return = paths - excluded
+      # filter out non existing files, so in case pattern is without '*' we don't do unnecessary work
+      to_return = to_return.select {|path|
+        Pathname.new(path).exist?
+      }
+      Hash[to_return.map {|path|
```

Comment:
Long lines in `Hash[]` seems hard to read.
This `map` can be moved to above line. `{}.select {}.map {}`.